### DISCUSSION TITLE
ESP32 internal temperature sensor: jumping values

### DISCUSSION THREAD

**Discussion by reimerp** [arendst/Tasmota#13026] [Q&A] [answered]
Hello all,

after installing tasmota to an esp32 (I only used esp8288 in the past) I noticed that the readings of the internal temperature sensors is jumping between two values:
`ESP32 Temperature	78.3 °C`
and a  couple of seconds later
`ESP32 Temperature	53.3 °C`

I used a fresh image with default settings. My Chip is a 
`ESP Chip Id	4568252 (ESP32-D0WDQ6)`

Unfortunately I have not tried to read the internal sensor with other images or with plain C code, but I do not think this is a tasmota issue.
Is my chip bad?
I read some documentation that the sensor is not present in newer chips, maybe there has been a design issue?

**Comment by Jason2866** [comment, marked as answer]
I just can say, the measurements i get from my ESP32-D0WDQ6 `s are always in the fiftys. No jumping values.

**Comment by reimerp** [comment]
I did a quick test with Micropython on that device: constant reading around 60°C, no jumping, so the chip seems ok.

**Reply by Jason2866** [reply]
Which Tasmota32 version/variant do you use?

**Comment by reimerp** [comment]
I am using the bluetooth version. The problem is gone when SO115 is OFF
When I activate it the temperature starts jumping....
I recorded the readings. The higher value seems to vary due to the load (70-75°C) while the lowest value (less frequent) seems to be constant 53,3°C.... Strange

**Comment by Jason2866** [comment]
Can not reproduce. Tested with a Sonoff Dual R3 (ESP32-D0WD-V3). A quiet new chip. BLE enabled. Temperature is between 53 and 58 °C.

**Comment by lestrozi** [comment]
I have exactly the same problem, also with invalid 53.3C readings
![Screenshot_20211225-220113_Chrome](https://user-images.githubusercontent.com/4925896/147394262-5f1867a9-89da-4a97-b3c2-ad8e959a90c8.jpg)

This image is for the values published via MQTT, but when I watch the Tasmota web UI, I have never seen the 53.3C value in there

Some googling suggests this happens when the sensor returns 128F (53.33C), which actually means the sensor is not available

Perhaps ignoring/retrying this value would be a quick fix

**Reply by sfromis** [reply]
You can get rid of that *unusable* number using the command `SetSensor127 0` (if you're not using the Hall Effect sensor), it has no value unless you like confusion or riddles. My experience is that the number does not represent anything real, certainly not the chip temperature (as checked with an ESP32 board without a metal shield covering the MCU chip).

**Comment by plewka** [comment]
The sensor is more intended to measure housekeeping of the esp32 than sensing its environment, in my understanding. It is coupled to the die and heats differently depending on activity/scanrate etc. I'd add an other sensor to measure temperature. The onboard sensors are a nice idea, but...yes, they are there. If you don't order special calibrated ESPs even the ADC may need some tuning.
